.buildkite/scripts/common/activate_service_account.sh

```diff
@@ -0,0 +1,75 @@
+#!/usr/bin/env bash
+
+set -euo pipefail
+
+source .buildkite/scripts/common/vault_fns.sh
+
+BUCKET_OR_EMAIL="${1:-}"
+
+if [[ -z "$BUCKET_OR_EMAIL" ]]; then
+  echo "Usage: $0 <bucket_name|email>"
+  exit 1
+elif [[ "$BUCKET_OR_EMAIL" == "-" ]]; then
```

Comment:
I think I was too naive then, and assumed everything would run in complete isolation. Also, the authentication was running as part of the `pre-command`.

Now the authentication is running right before we want to use the buckets, so presumably when we're running the pipelines, and not the upload - I hope that's less of a risk now, as the agents are throwaway (on the new infra, I think the agents are one-use).

Regardless, I can try to set up a logout mechanism with redundancy/resiliency in the post-commands. E.g.: see if there's an impersonated account, remove the impersonation config; see if the `gcloud` identity is the one for the service-account proxy, then log-out.

Reply:
This is printed when we're trying to revoke service-account tokens:
```
WARNING: [kibana-ci-sa-proxy@elastic-kibana-ci.iam.gserviceaccount.com] appears to be a service account. Service account tokens cannot be revoked, but they will expire automatically. To prevent use of the service account token earlier than the expiration, delete or disable the parent service account. To explicitly delete the key associated with the service account use `gcloud iam service-accounts keys delete` instead`.

Revoked credentials:
--
  | - kibana-ci-sa-proxy@elastic-kibana-ci.iam.gserviceaccount.com
  | Credentialed Accounts
  | ACTIVE  ACCOUNT
  | *       elastic-buildkite-agent@elastic-kibana-ci.iam.gserviceaccount.com
  |  
```
So it does reset the identity to the previous one - I'll try to avoid the warning in [c274d87](https://github.com/elastic/kibana/pull/174756/commits/c274d87dc34b6834bb77769a6cb2eaf832fe3312)